homeassistant/components/imap/coordinator.py

```diff
@@ -496,7 +496,16 @@ async def _async_wait_push_loop(self) -> None:
                 await self.imap_client.wait_server_push()
                 self.imap_client.idle_done()
                 async with asyncio.timeout(10):
-                    await idle
+                    try:
+                        await idle
+                    except asyncio.CancelledError:
+                        _LOGGER.debug(
+                            "Connection canceled with %s (will attempt to reconnect after %s s)",
+                            self.config_entry.data[CONF_SERVER],
+                            BACKOFF_TIME,
+                        )
+                        await self._cleanup()
+                        await asyncio.sleep(BACKOFF_TIME)
```

Comment:
This is still swallowing the CancelledError because it never raises.

Reply:
It might actually be ok because `idle` is a future here and if the current task is cancelling it should still raise on `await self._cleanup()`

Reply:
Probably do this at the top of the except block to be sure it doesn't swallow
```
		if asyncio.current_task().cancelling() > 0:
			raise
```

Reply:
Should we exit the loop?

Reply:
If the current task is cancelling and we raise, the whole coro should end